owncloudData/src/test/java/com/owncloud/android/data/transfers/implementation/OCLocalTransferDataSourceTest.kt

```diff
@@ -0,0 +1,365 @@
+    }
+
+    @Test
+    fun deleteTransferById() {
+        every {
+            transferDao.deleteTransferWithId(id)
+        } returns Unit
+
+        ocLocalTransferDataSource.deleteTransferById(id)
+
+        verify(exactly = 1) {
+            transferDao.deleteTransferWithId(id)
+        }
+    }
+
+    @Test
+    fun deleteAllTransfersFromAccount() {
+        every {
+            transferDao.deleteTransfersWithAccountName(OC_ACCOUNT_NAME)
+        } returns Unit
+
+        ocLocalTransferDataSource.deleteAllTransfersFromAccount(OC_ACCOUNT_NAME)
+
+        verify(exactly = 1) {
+            transferDao.deleteTransfersWithAccountName(OC_ACCOUNT_NAME)
+        }
+    }
+
+    @Test
+    fun `getTransferById returns a OCTransfer`() {
+        every {
+            transferDao.getTransferWithId(any())
+        } returns ocTransfer.toEntity()
+
+        val actualResult = ocLocalTransferDataSource.getTransferById(id)
+
+        assertEquals(ocTransfer, actualResult)
+
+        verify(exactly = 1) {
+            transferDao.getTransferWithId(id)
+        }
+    }
+
+    @Test
+    fun `getAllTransfers returns a list of OCTransfer`() {
+
+        every {
+            transferDao.getAllTransfers()
+        } returns listOf(transferEntity)
+
+        val actualResult = ocLocalTransferDataSource.getAllTransfers()
+
+        assertEquals(listOf(transferEntity.toModel()), actualResult)
+
+        verify(exactly = 1) {
+            transferDao.getAllTransfers()
+        }
+    }
+
+    @Test
+    fun `getAllTransfersAsStream returns a flow of list of OCTransfer with transferStatus in progress`() = runBlocking {
+
+        val transferEntityInProgress: OCTransferEntity = ocTransfer.toEntity()
+        transferEntityInProgress.status = 0
+
+        val transferEntityQueue: OCTransferEntity = ocTransfer.toEntity()
+        transferEntityQueue.status = 1
+
+        val transferEntityFailed: OCTransferEntity = ocTransfer.toEntity()
+        transferEntityFailed.status = 2
+
+        val transferEntitySucceeded: OCTransferEntity = ocTransfer.toEntity()
+        transferEntitySucceeded.status = 3
+
+        val transferListRandom = listOf(transferEntityQueue, transferEntityFailed, transferEntityInProgress, transferEntitySucceeded)
+
+
+        val transferQueue = ocTransfer.copy()
+        transferQueue.status =  TransferStatus.TRANSFER_QUEUED
+
+        val transferFailed = ocTransfer.copy()
+        transferFailed.status =  TransferStatus.TRANSFER_FAILED
+
+        val transferSucceeded = ocTransfer.copy()
+        transferSucceeded.status =  TransferStatus.TRANSFER_SUCCEEDED
+
+        val transferListOrdered = listOf(ocTransfer, transferQueue, transferFailed, transferSucceeded)
+
+        every {
+            transferDao.getAllTransfersAsStream()
+        } returns flowOf(transferListRandom)
+
+        val actualResult = ocLocalTransferDataSource.getAllTransfersAsStream().first().map { it }
+        val expectedStatusList = transferListOrdered.map { it }
```

Comment:
These `map`s don't make sense, you already have a list and `map { it }` just keeps the elements as they are (no changes)

Reply:
Done